src/core/qgsmaprendererjob.cpp

```diff
@@ -817,7 +817,17 @@ void QgsMapRendererJob::composeSecondPass( LayerRenderJobs &secondPassJobs, Labe
       // Only retain parts of the second rendering that are "inside" the mask image
       QPainter *painter = job.context.painter();
       painter->setCompositionMode( QPainter::CompositionMode_DestinationIn );
-      painter->drawImage( 0, 0, *maskImage );
+
+      //Create an "alpha binarized" image of the maskImage to :
+      //* Eliminate antialiasing artefact
+      //* Avoid applying mask opacity to elements under the mask but not masked
+      QImage maskBinAlpha = maskImage->createMaskFromColor( 0 );
+      maskBinAlpha.setColorCount( 0 );
```

Comment:
Is this necessary? Seems like it would be redundant given the call to setColorTable below...

Reply:
I had some issues at one point with color count being incremented but you are right it seems it is now useless. I removed it.